buildSrc/src/main/kotlin/Versions.kt

```diff
@@ -0,0 +1,61 @@
+import kotlin.String
+import org.gradle.plugin.use.PluginDependenciesSpec
+import org.gradle.plugin.use.PluginDependencySpec
+
+/**
+ * Generated by https://github.com/jmfayard/buildSrcVersions
+ *
+ * Find which updates are available by running
+ *     `$ ./gradlew buildSrcVersions`
+ * This will only update the comments.
+ *
+ * YOU are responsible for updating manually the dependency version.
+ */
+object Versions {
```

Comment:
As said before, instead of having a large file with the versions of all the libraries, I'd put the versions as a companion object inside of each file. If they are `object` you can not use companion object inside but you could do something like : 

```
object AndroidDependencies {
const val constraintLayout : ......
}

private object AndroidDependenciesVersions{
const val CONSTRAINT_LAYOUT_VERSION : ....
}
```

Reply:
Same answer as before, for now this is the best workaround that I found until Android Studio gives full support to the Kotlin DSL